pkg/adapters/grpc/document_grpc.go

```diff
@@ -30,7 +32,19 @@ type DocumentServiceServer struct {
 	documentPlugin document.DocumentService
 }
 
+func (s *DocumentServiceServer) checkPluginRegistered() error {
+	if s.documentPlugin == nil {
+		return status.Errorf(codes.Unimplemented, "Document plugin not registered")
+	}
+
```

Comment:
Could be worth creating an error helper function in the package ```utils.go``` to help standardize these messages:

```
func PluginNotRegisteredError(name string) error {
	return status.Errorf(codes.Unimplemented, "%s plugin not registered", name)
}
```

Reply:
Done